wp-includes/sqlite-ast/class-wp-sqlite-driver.php

```diff
@@ -3358,25 +3358,26 @@ private function translate_string_literal( WP_Parser_Node $node ): string {
 		 * Removing them would damage the serialized data.
 		 *
 		 * There is no way to store null bytes using a string literal, so we
-		 * need to split the string and concatenate null bytes with its parts.
+		 * need to pass the value as a HEX string and cast it back to TEXT.
 		 * This will convert literals will null bytes to expressions.
 		 *
 		 * Alternatively, we could replace string literals with parameters and
 		 * pass them using prepared statements. However, that's not universally
 		 * applicable for all string literals (e.g., in default column values).
 		 *
+		 * We can't use the "part1 || CHAR(0) || part2 || ..." syntax, because
+		 * with a large number of null bytes, SQLite throws the following error:
+		 *
+		 *   SQLSTATE[HY000]:
+		 *   General error: 1 Expression tree is too large (maximum depth 1000)
+		 *
 		 * See:
 		 *   https://www.sqlite.org/nulinstr.html
 		 */
-		$parts = array();
-		foreach ( explode( "\0", $value ) as $segment ) {
-			// Escape and quote each segment.
-			$parts[] = "'" . str_replace( "'", "''", $segment ) . "'";
-		}
-		if ( count( $parts ) > 1 ) {
-			return '(' . implode( ' || CHAR(0) || ', $parts ) . ')';
+		if ( strpos( $value, "\0" ) !== false ) {
+			return sprintf( "CAST(x'%s' AS TEXT)", bin2hex( $value ) );
 		}
-		return $parts[0];
+		return sprintf( "'%s'", str_replace( "'", "''", $value ) );
```

Comment:
don't we have some helper for escaping string values?

Reply:
@adamziel Ahh, good point. This is a string value so the native PDO\Sqlite::quote() should work here. Pushed in https://github.com/WordPress/sqlite-database-integration/pull/274/commits/455a139d23221ba8820ad46090212a4696b4285f.